daemon/export.go

```diff
@@ -42,23 +43,48 @@ func (daemon *Daemon) ContainerExport(ctx context.Context, name string, out io.W
 	return nil
 }
 
-func (daemon *Daemon) containerExport(ctx context.Context, container *container.Container, out io.Writer) error {
-	err := daemon.imageService.PerformWithBaseFS(ctx, container, func(basefs string) error {
-		archv, err := chrootarchive.Tar(basefs, &archive.TarOptions{
-			Compression: archive.Uncompressed,
-			IDMap:       daemon.idMapping,
-		}, basefs)
+func (daemon *Daemon) containerExport(ctx context.Context, ctr *container.Container, out io.Writer) error {
+	rwlayer := ctr.RWLayer
+	if rwlayer == nil {
+		return fmt.Errorf("container %s has no rootfs", ctr.ID)
+	}
+
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	default:
+	}
+
+	basefs, err := rwlayer.Mount(ctr.GetMountLabel())
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err := rwlayer.Unmount()
 		if err != nil {
-			return err
+			log.G(ctx).WithError(err).WithField("container", ctr.ID).Warn("Failed to unmount container RWLayer after export")
```

Comment:
Perhaps use `WithFields` here;


```suggestion
			log.G(ctx).WithFields(log.Fields{"error": err, "container": ctr.ID}).Warn("Failed to unmount container RWLayer after export")
```